pkg/build/builder/dockerutil.go

```diff
@@ -62,15 +73,35 @@ func pushImage(client DockerClient, name string, authConfig docker.AuthConfigura
 	if glog.V(5) {
 		opts.OutputStream = os.Stderr
 	}
-	var err error
+
 	var retriableError = false
 
+	eventWatch, err := eventClient.Watch(
```

Comment:
> I'm assuming the watch returns just additions since the listing started.

I don't think so.  You're not specifying a `resourceVersion`, so I'm pretty sure you fall into the: 



> Empty string means "send me add events for all existing things and any future watch events"